base-tailwind.config.js

```diff
@@ -0,0 +1,96 @@
+import {
+    scopedPreflightStyles,
+    isolateInsideOfContainer,
+} from 'tailwindcss-scoped-preflight';
+
+const rootClass = '.dokan-layout'; //We will use this class to scope the styles.
+
+/** @type {import('tailwindcss').Config} */
+const baseConfig = {
+    important: rootClass,
+    content: [ './src/**/*.{js,jsx,ts,tsx}', '!./**/*.asset.php' ],
+    theme: {
+        extend: {
+            backgroundColor: {
+                dokan: {
+                    sidebar: {
+                        DEFAULT:
+                            'var(--dokan-sidebar-background-color, #F05025)',
+                        hover: 'var(--dokan-sidebar-hover-background-color, #F05025)',
+                    },
+                    btn: {
+                        DEFAULT:
+                            'var(--dokan-button-background-color, #F05025)',
+                        hover: 'var(--dokan-button-hover-background-color, #F05025)',
+                        secondary: {
+                            DEFAULT:
+                                'var(--dokan-button-secondary-background-color, var(--dokan-button-text-color))',
+                            hover: 'var(--dokan-button-secondary-hover-background-color, var(--dokan-button-background-color))',
+                        },
+                        tertiary: {
+                            DEFAULT:
+                                'var(--dokan-button-tertiary-background-color, #00000000)',
+                            hover: 'var(--dokan-button-tertiary-hover-background-color, var(--dokan-button-text-color))',
+                        },
+                    },
+                },
+            },
+            textColor: {
+                dokan: {
+                    sidebar: {
+                        DEFAULT: 'var(--dokan-sidebar-text-color, #CFCFCF)',
+                        hover: 'var(--dokan-sidebar-hover-text-color, #ffffff)',
+                    },
+                    btn: {
+                        DEFAULT: 'var(--dokan-button-text-color, #ffffff)',
+                        hover: 'var(--dokan-button-hover-text-color, #ffffff)',
+                        secondary: {
+                            DEFAULT:
+                                'var(--dokan-button-secondary-text-color, var(--dokan-button-background-color))',
+                            hover: 'var(--dokan-button-secondary-hover-text-color, var(--dokan-button-text-color))',
```

Comment:
What will be the default color if the css variables are not declared/assigned.

CC: @shohag121 vai.